### DISCUSSION TITLE
Is Positron going to support Linux ARM64 distributions?

### DISCUSSION THREAD

**Discussion by andresrcs** [posit-dev/positron#3743] [Q&A] [answered]
Since RStudio is building for arm64 already on their dailies builds, will you consider on the future build Positron for arm64?

**Comment by wesm** [comment, marked as answer]
Yes, Positron will eventually support this but "when" is TBD. As part of this, we will have to test and create builds of the Ark R kernel (https://github.com/posit-dev/ark) -- interested arm64 Linux users are welcome to make contributions to help!

**Comment by pat-s** [comment]
`ark` has meanwhile support for linux arm64 and the most recent Workbench builds now include Positron in "preview". Yet I noticed that there are no Positron builds for Linux arm64 yet. Any plans to do so soonish or are there additional unknown blockers?

**Reply by jmcphers** [reply]
Aside from the paperwork to set up the additional build matrix, we also need an arm64 build of the kernel supervisor (which sits on the other side of ark's zeromq sockets), which in turn requires a Linux arm64 build host. Unlike ark itself, the kernel supervisor cannot be cross-compiled on an x86 host. 

Positron and its components primarily build on Github Actions, which only added support for arm64 Linux hosts a few months ago. We do plan to take advantage of this infrastructure soon, but it may be a bit since (even now) arm64 Linux hosts are not available as public images from Github.

**Reply by pat-s** [reply]
Thanks for the quick answer @jmcphers 

With the supervisor you are referring to https://github.com/posit-dev/kallichore-builds?

Would you make use of them if such binaries would be available from an external source? I could build them on my own arm64 runners in an automated way using a mirror repo until you're ready to go with your arm64 runners.
